task4/src/components/movie/movie.js

```diff
@@ -1,21 +1,44 @@
 import React from 'react';
 import './movie.scss';
 import PropTypes from 'prop-types';
+import EditMovie from '../editMovie/editMovie';
 
-function Movie(props) {
-  const {poster = '', title = '', genre = '', year = ''} = props.movie;
-  return (
-    <div className="movie">
-      <img src={poster}/>
-      <div className="movie-info">
-        <h3>
-          {title}
-          <span>{genre}</span>
-        </h3>
-        <span className="year">{year}</span>
+class Movie extends React.Component{
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      movie: props.movie,
+      showModal: false,
+    };
+  }
+
+  toggleModal(value) {
+    console.log(value || !this.state.showModal);
+    this.setState({
+      showModal: value || !this.state.showModal
+    });
+  }
+
+  render() {
```

Comment:
to avoid always write this.pros... or this.state... you can do something like this

```
render(){
    const {movie} = this.props;
    const {shoeModal} = this.state;
...
```